datafusion/common/src/table_reference.rs

```diff
@@ -35,7 +35,38 @@ impl<'a> std::fmt::Display for ResolvedTableReference<'a> {
     }
 }
 
-/// Represents a path to a table that may require further resolution
+/// [`TableReference`]s represent a multi part identifier (path) to a
+/// table that may require further resolution.
+///
+/// # Creating [`TableReference`]
+///
+/// When converting strings to [`TableReference`]s, the string is
```

Comment:
The SQL parsing behavior added in https://github.com/apache/arrow-datafusion/pull/5343 is somewhat surprising so I think it should be well documented